tests/integration/cattletest/core/test_svc_upgrade.py

```diff
@@ -863,3 +863,35 @@ def test_rollback_id(context, client, super_client):
     expose_map = maps[0]
     c2 = super_client.reload(expose_map.instance())
     assert c1.uuid == c2.uuid
+
+
+def test_in_service_upgrade_port_mapping(context, client, super_client):
+    env = client.create_environment(name=random_str())
+    env = client.wait_success(env)
+    image_uuid = context.image_uuid
+    launch_config = {"imageUuid": image_uuid, 'ports': ['80']}
+    secondary1 = {"imageUuid": image_uuid, "name": "secondary1",
```

Comment:
Added test for port with protocol and also a new public:private port pair in upgrade.